src/algorithm/MEMCPY-Cuda.cpp

```diff
@@ -0,0 +1,218 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2017-22, Lawrence Livermore National Security, LLC
+// and RAJA Performance Suite project contributors.
+// See the RAJAPerf/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "MEMCPY.hpp"
+
+#include "RAJA/RAJA.hpp"
+
+#if defined(RAJA_ENABLE_CUDA)
+
+#include "common/CudaDataUtils.hpp"
+
+#include <iostream>
+
+namespace rajaperf
+{
+namespace algorithm
+{
+
+#define MEMCPY_DATA_SETUP_CUDA \
+  allocAndInitCudaDeviceData(x, m_x, iend); \
+  allocAndInitCudaDeviceData(y, m_y, iend);
+
+#define MEMCPY_DATA_TEARDOWN_CUDA \
+  getCudaDeviceData(m_y, y, iend); \
+  deallocCudaDeviceData(x); \
+  deallocCudaDeviceData(y);
+
+template < size_t block_size >
+__launch_bounds__(block_size)
+__global__ void memcpy(Real_ptr x, Real_ptr y,
+                       Index_type iend)
+{
+  Index_type i = blockIdx.x * block_size + threadIdx.x;
+  if ( i < iend ) {
+    MEMCPY_BODY;
+  }
+}
+
+
+void MEMCPY::runCudaVariantMemcpy(VariantID vid)
+{
+  const Index_type run_reps = getRunReps();
+  const Index_type ibegin = 0;
+  const Index_type iend = getActualProblemSize();
+
+  MEMCPY_DATA_SETUP;
+
+  if ( vid == Base_CUDA ) {
+
+    MEMCPY_DATA_SETUP_CUDA;
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
+
+      cudaErrchk( cudaMemcpyAsync(MEMCPY_STD_ARGS, cudaMemcpyDefault, 0) );
+
+    }
+    stopTimer();
+
+    MEMCPY_DATA_TEARDOWN_CUDA;
+
+  } else if ( vid == RAJA_CUDA ) {
+
+    MEMCPY_DATA_SETUP_CUDA;
+
+    camp::resources::Cuda res = camp::resources::Cuda::get_default();
+
+    startTimer();
+    for (RepIndex_type irep = 0; irep < run_reps; ++irep) {
```

Comment:
Why is this considered a RAJA variant?

Reply:
This uses a camp resource to do the operation and the camp resource uses the native library operation. So I interpreted this as the RAJA abstraction layer's way to use the native library operation.

Reply:
Makes sense, I guess.